src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -570,6 +570,22 @@ public static void CheckCanGenerateEEType(NodeFactory factory, TypeDesc type)
                     // TODO: validate constraints
                 }
 
+                // Check the type doesn't have bogus MethodImpls or overrides and we can get the finalizer.
+                defType.GetFinalizer();
+
+                const int FieldOffsetMax = (1 << 27) - 1; // FIELD_OFFSET_MAX in CoreCLR
+                const int FieldOffsetLastRealOffset = FieldOffsetMax - 6; // FIELD_OFFSET_LAST_REAL_OFFSET in CoreCLR
```

Comment:
This field offset check looks like questionable bug-for-bug compatibility to me. I do not think we should have it.